File: E_Commerce/E_Commerce/E_Commerce.Dal/Entities/Payment.cs
using E_Commerce.Dal.Enums;

namespace E_Commerce.Dal.Entities;

public class Payment
{
    public long PaymentId { get; set; }
    public PaymentMethod PaymentMethod { get; set; }
    public PaymentStatus PaymentStatus { get; set; }
    public decimal PaidAmount { get; set; }
    public DateTime PaidAt { get; set; }
    public long OrderId { get; set; }
    public Order Order { get; set; }
}

File: Client/src/app/services/models/Payment.ts
import { OrderModel } from "./Order";

export enum PaymentMethod {
  Click = 'Click',
  PayPal = 'PayPal',
  Payme = 'Payme',
  Uzum = 'Uzum',
  CashPayment = 'CashPayment'
}

export enum PaymentStatus {
  Pending = 'Pending',
  Completed = 'Completed',
  Failed = 'Failed',
  Refunded = 'Refunded',
  
}

// --- CLASS DEFINITION ---

export class PaymentModel {
  paymentId: number = 0; // long in C# maps to number in TypeScript
  paymentMethod: PaymentMethod = PaymentMethod.Click; // Default to a specific method
  paymentStatus: PaymentStatus = PaymentStatus.Pending; // Default to Pending
  paidAmount: number = 0; // decimal in C# maps to number in TypeScript
  paidAt: Date = new Date(); // DateTime in C# maps to Date in TypeScript
  orderId: number = 0; // long in C# maps to number in TypeScript
  order?: OrderModel; // Optional, as it might be lazy-loaded or not always included}
}
